subprojects/scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java

```diff
@@ -147,7 +148,8 @@ public void compile() {
      * @since 7.2
      */
     @Incubating
-    @Internal
+    @Nested
+    @Optional
     public Property<JavaLauncher> getJavaLauncher() {
```

Comment:
You added the test for the events. Though what we'd need now as well is that the task is out-of-date when the toolchain changes.

Reply:
Good catch, I will add that

Reply:
Done